arangod/Replication2/StateMachines/Document/MaintenanceActionExecutor.h

```diff
@@ -70,12 +72,17 @@ class MaintenanceActionExecutor : public IMaintenanceActionExecutor {
       ShardID shard, std::shared_ptr<VPackBuilder> const& properties,
       std::shared_ptr<methods::Indexes::ProgressTracker> progress)
       -> Result override;
+  auto executeDropIndex(ShardID shard, velocypack::SharedSlice index)
+      -> Result override;
   void addDirty() override;
 
  private:
   GlobalLogIdentifier _gid;
   MaintenanceFeature& _maintenanceFeature;
   ServerID _server;
+
+  // The vocbase reference remains valid for the lifetime of the executor.
+  // Replicated logs are stopped before the vocbase is marked as dropped.
```

Comment:
Ah so this means someone outside holds the DatabaseGuard?
if that is true ignore my above comments on the DatabaseGuard.

Reply:
There is no need for a `DatabaseGuard`. I have added it "out of reflex" before, but we made sure that, when the database is being dropped or the system is shutting down, the vocbase reference only becomes invalid after the replicated logs have been shut down.
This happens in `DatabaseFeature::dropDatabase()`. There, `shutdownReplicatedLogs()` is called before the vocbase is marked as dropped.

Reply:
I am not against using one though, just for the safety of it, but I would rather do it for all the helpers then, in a way which does not interfere with unit tests. I can do this in a later refactoring.

Reply:
Okay, I agree we are protected against dropping, so no Guard required.